### PR TITLE
fix(kafka source): Fix integration test failure

### PR DESCRIPTION
**Pull request by bruceg**

### CODE REVIEW

File: src/sources/kafka.rs [hunk 1]
```diff
@@ -588,6 +588,9 @@ mod integration_test {
                 acknowledgements,
             ));
             let events = collect_n(rx, 10).await;
+            // Wait a beat to let the finalization task collect the
+            // status receivers sent to it.
+            tokio::task::yield_now().await;
```
Comment (fuchsnj) [inline src/sources/kafka.rs:593]: Is this because the single_threaded executor is being used? Should that be switched to multi-threaded?

File: src/sources/kafka.rs [hunk 2]
```diff
@@ -588,6 +588,9 @@ mod integration_test {
                 acknowledgements,
             ));
             let events = collect_n(rx, 10).await;
+            // Wait a beat to let the finalization task collect the
```
Comment (fuchsnj) [inline src/sources/kafka.rs:591]: Typo I think?
```suggestion
            // Wait a bit to let the finalization task collect the
```

File: src/sources/kafka.rs (hunk 1 above)
Comment (bruceg) [reply src/sources/kafka.rs:593]: No, it's because there's a fundamental race between the batch notifiers being sent to the finalizer set and the shutdown signal. It's always been there, but the previous change made it (more) repeatable. I added debug statements to the integration test and saw that sometimes just one ack would get through, sometimes 3, but always less than the full 10, so even with this single-threaded executor it is non-deterministic. The yield (and I have a follow up patch I'll submit separately) just ensures that the channel empties before returning to this point.

File: src/sources/kafka.rs (hunk 2 above)
Comment (bruceg) [reply src/sources/kafka.rs:591]: Yeah, the "beat" was intentional but probably not helpful. I'll reword it to clarify.

File: src/sources/kafka.rs (hunk 1 above)
Comment (fuchsnj) [reply src/sources/kafka.rs:593]: ok, I'm just a bit concerned since `yield_now()` doesn't really guarantee anything meaningful that you should rely on, but I"m fine if this is a temporary fix since it sounds like you are following up with something else

**Review by fuchsnj — APPROVED** [review]

File: src/sources/kafka.rs (hunk 1 above)
Comment (bruceg) [reply src/sources/kafka.rs:593]: Hmm, the docs say that:
> The current task will be re-added as a pending task at the back of the pending queue. Any other pending tasks will be scheduled.

Knowing that the tests are run on a single-threaded runtime, this reads that it will cause any ready work to be immediately scheduled, which in this case I know will be present, and then rerun once that is completed or becomes pending as well.

Comment (fuchsnj) [reply src/sources/kafka.rs:593]: It also says
> In particular, the runtime may choose to poll the task that just ran
/// `yield_now()` again immediately without polling any other tasks first

which I read as basically saying `yield_now()` might not do anything at all.
